File: program/program/implement-jump-search/implement_jump_search.py
import math


def jump_search(arr, x, n):
    block_size = math.sqrt(n)

    while arr[int(min(block_size, n) - 1)] < x:
        prev = block_size
        block_size += math.sqrt(n)
        if prev >= n:
            return -1

    while arr[int(prev)] < x:
        prev += 1

        if prev == min(block_size, n):
            return -1

    if arr[int(prev)] == x:
        return int(prev)

    return -1


if __name__ == '__main__':
    list = [1, 2, 3, 4, 5]
    value = 4
    n = len(list)
    print(jump_search(list, value, n))

File: program/program/implement-jump-search/ImplementJumpSearch.rb
def jump_search(arr, x)
  n = arr.size
  block_size = Math.sqrt(n).to_i

  prev = 0
  while arr[[block_size, n].min - 1] < x
    prev = block_size
    block_size += Math.sqrt(n).to_i
    return -1 if prev >= n
  end

  while arr[prev] < x
    prev += 1
    return -1 if prev == [block_size, n].min
  end

  return prev if arr[prev] == x

  -1
end

# Example usage
arr = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
x = 7
result = jump_search(arr, x)

if result != -1
  puts "Element found at index #{result}"
else
  puts "Element not found in the array"
end
